docker.go

```diff
@@ -507,15 +503,28 @@ func (docker *dockerConfig) refreshImage(image string) {
 	log.Println()
 }
 
-func loginToECR(account string, region string) {
-	svc := ecr.New(ecr.Options{Region: region})
+func (docker *dockerConfig) tryLoginToECR(image string) error {
+	matches, _ := reutils.MultiMatch(image, reECR)
+	account, accountOk := matches["account"]
+	if !accountOk {
+		return fmt.Errorf("%v is not an ECR image", image)
+	}
+	log.Debugf("Failed to pull %v. It is an ECR image, trying again after login to AWS ECR.", image)
```

Comment:
This log line suggests we failed to do something, but we never really tried before logging it?

Reply:
I think we do in fact. From line 494, we try to get UpdateCmd which I believe tries to get the image info ( so kind of pulling) and ends here if it failed.

Reply:
Yeah, I see it outside the method, I think I'd put the log line there then though. Here in this context, there's no previous action attempted.

Reply:
I changed the error management to make it more explicit and less verbose